### ISSUE TITLE
Core Profiler plugin selection does not have a visible hover state for keyboard navigation

### ISSUE THREAD

**Issue by crunnells** [woothemes/woocommerce#38753]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In the plugin selection step of the Core Profiler, all of the plugins checkboxes are pre-checked by default. If a person is trying to navigate this list with a keyboard, there is _very little_ visual indication of which checkbox is selected. This is partially due to the `:checked` styling, which has an outline around it. This styling might be better used to indicate `:hover`, `:focus`, and/or `:active` states instead. The `:checked` styling by itself is visually obvious (checkmark and checkbox background color).

<img width="1020" alt="Screen Shot 2023-06-15 at 2 54 23 PM" src="https://github.com/woocommerce/woocommerce/assets/4297811/3ff4b6f7-1911-46cd-bf91-f765db39bcd1">
(One of these is selected, but which one?)

### Expected behavior

When navigating this page via keyboard, it should be visually obvious which checkbox is selected.

### Actual behavior

Checkbox styling for `:focus` is _very_ subtle and difficult to visually differentiate between elements.

### Steps to reproduce

1. Install and activate WooCommerce on a brand new site
2. Install the WooCommerce Beta Tester plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper
4. Enable the 'core-profiler' feature flag
5. Visit /wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard
6. Proceed through the first three steps until you get to the "Get a boost with our free features" page.
7. Use the keyboard (tab key) to navigate through the elements of the page.

### WordPress Environment

n/a, this is present in the new Core Profiler feature

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by chihsuan** [comment]
Closing by https://github.com/woocommerce/woocommerce/issues/50151.
